File: Assets/Scripts/Buttons.js
#pragma strict
//Suggestive text >:)
public var Cant : GameObject;
public var RedCant : GameObject;
public var Cant2 : GameObject;
public var Cant3 : GameObject;
public var Cant4 : GameObject;
public var Cant5 : GameObject;
public var Cant6 : GameObject;
public var Cant7 : GameObject;
//Buttons
public var levels5 : GameObject;
public var levels4 : GameObject;
public var levels3 : GameObject;
public var levels2 : GameObject;
//Blockers
public var Blocker2 : GameObject;
public var Blocker3 : GameObject;
public var Blocker4 : GameObject;
public var Blocker5 : GameObject;
var Scream : int;

function Start () {
	if (Application.loadedLevelName == "Level") {
		if (PlayerPrefs.GetInt("DoneLevel1") == 0) {
			Blocker2.SetActive(true);
			Blocker3.SetActive(true);
			Blocker4.SetActive(true);
			Blocker5.SetActive(true);
		}else if (PlayerPrefs.GetInt("DoneLevel2") == 0) {
			Blocker3.SetActive(true);
			Blocker4.SetActive(true);
			Blocker5.SetActive(true);
		}else if (PlayerPrefs.GetInt("DoneLevel3") == 0) {
			Blocker4.SetActive(true);
			Blocker5.SetActive(true);
		}else if (PlayerPrefs.GetInt("DoneLevel4") == 0) {
			Blocker5.SetActive(true);
		}else if (PlayerPrefs.GetInt("DoneLevel5") == 0) {
			//nothing shall pass
		}
	}else if (Application.loadedLevelName == "RLevel") {		//Reversed Level pack 1
		if (PlayerPrefs.GetInt("DoneRLevel1") == 0) { 
			Blocker2.SetActive(true);
			Blocker3.SetActive(true);
			Blocker4.SetActive(true);
			Blocker5.SetActive(true);
		}else if (PlayerPrefs.GetInt("DoneRLevel2") == 0) {
			Blocker3.SetActive(true);
			Blocker4.SetActive(true);
			Blocker5.SetActive(true);
		}else if (PlayerPrefs.GetInt("DoneRLevel3") == 0) {
			Blocker4.SetActive(true);
			Blocker5.SetActive(true);
		}else if (PlayerPrefs.GetInt("DoneRLevel4") == 0) {
			Blocker5.SetActive(true);
		}
	}
	
}

function update () {
	Screen.sleepTimeout = SleepTimeout.NeverSleep;
}

function QuitGame ()  {
	Debug.Log ("Game is exiting...");
	Application.Quit ();
}

function MainMenu() {
	Application.LoadLevel("MainMenu");
	Time.timeScale = 1;
}

function Playgame() {
	if(PlayerPrefs.HasKey("savedLevel")){
   		// there is a saved Level, load that one
   		Application.LoadLevel(PlayerPrefs.GetInt("savedLevel"));
	} else{
   		// no saved Level, load the first one
   		Application.LoadLevel("Level1");
 	}
}

function Tutorial() {
	Application.LoadLevel("Tutorial");
}

//All the scene selection stuff
function levelselectoor() {
	Application.LoadLevel("LevelSelect");
}
function levels() {
	Application.LoadLevel("Level");
}
function levelsR() {
	Application.LoadLevel("RLevel");
}

//Level Pack 1
function Level1() {
	Application.LoadLevel("Level1");
}
function Level2() {
	if (PlayerPrefs.GetInt("DoneLevel1") == 1) {
		Application.LoadLevel("Level2");
	}else {
		Warning();
	}
}
function Level3() {
	if (PlayerPrefs.GetInt("DoneLevel2") == 1) {
		Application.LoadLevel("Level3");
	}else {
		Warning();
	}
}
function Level4() {
	if (PlayerPrefs.GetInt("DoneLevel3") == 1) {
		Application.LoadLevel("Level4");
	}else {
		Warning();
	}
}
function Level5() {
	if (PlayerPrefs.GetInt("DoneLevel4") == 1) {
		Application.LoadLevel("Level5");
	}else {
		Warning();
	}
}
//Reverse Level pack
function RLevel1() {
	Application.LoadLevel("RLevel1");
}
function RLevel2() {
	if (PlayerPrefs.GetInt("RLevel1") == 1) {
		Application.LoadLevel("RLevel2");
	}else {
		Warning();
	}
}
function RLevel3() {
	if (PlayerPrefs.GetInt("RLevel2") == 1) {
		Application.LoadLevel("RLevel3");
	}else {
		Warning();
	}
}
function RLevel4() {
	if (PlayerPrefs.GetInt("RLevel3") == 1) {
		Application.LoadLevel("RLevel4");
	}else {
		Warning();
	}
}
function RLevel5() {
	if (PlayerPrefs.GetInt("RLevel5") == 1) {
		Application.LoadLevel("RLevel5");
	}else {
		Warning();
	}
}


function Warning() {
	Scream += 1;
	if (Scream == 1) {
		Cant.SetActive (true);
	}else if (Scream == 2) {
		Cant.SetActive (false);
		RedCant.SetActive (true);
	}else if (Scream == 3) {
		RedCant.SetActive (false);
		yield WaitForSeconds (.5);
		RedCant.SetActive (true);
	}else if (Scream == 4) {
		Cant2.SetActive(true);
	}else if (Scream == 5) {
		Cant3.SetActive(true); //ugh
	}else if (Scream == 6) {
		Cant4.SetActive(true); //making me mad
	}else if (Scream == 7) {
		Cant5.SetActive(true); //your asking for it
	}else if (Scream == 8) {
		Cant5.SetActive(false);
		Cant4.SetActive(false);
		Cant3.SetActive(false);
		Cant2.SetActive(false);
		RedCant.SetActive(false);
		Cant6.SetActive(true); //one more time
	}else if (Scream == 9) {
		//NUCLEAR
		if (PlayerPrefs.GetInt("DoneLevel1") == 0) {
			levels5.SetActive(false);
			levels4.SetActive(false);
			levels3.SetActive(false);
			levels2.SetActive(false);
		}else if (PlayerPrefs.GetInt("DoneLevel2") == 0) {
			levels5.SetActive(false);
			levels4.SetActive(false);
			levels3.SetActive(false);
		}else if (PlayerPrefs.GetInt("DoneLevel3") == 0) {
			levels5.SetActive(false);
			levels4.SetActive(false);
		}else if (PlayerPrefs.GetInt("DoneLevel4") == 0) {
			levels5.SetActive(false);
		}else if (PlayerPrefs.GetInt("DoneRLevel1") == 0) { //Reversed Level Pack 1
			levels5.SetActive(false);
			levels4.SetActive(false);
			levels3.SetActive(false);
			levels2.SetActive(false);
		}else if (PlayerPrefs.GetInt("DoneRLevel2") == 0) {
			levels5.SetActive(false);
			levels4.SetActive(false);
			levels3.SetActive(false);
		}else if (PlayerPrefs.GetInt("DoneRLevel3") == 0) {
			levels5.SetActive(false);
			levels4.SetActive(false);
		}else if (PlayerPrefs.GetInt("DoneRLevel4") == 0) {
			levels5.SetActive(false);
		}
		Cant6.SetActive(false);
		Cant7.SetActive(true);
		Blocker5.SetActive(false);
		Blocker4.SetActive(false);
		Blocker3.SetActive(false);
		Blocker2.SetActive(false);
	}
}

File: Assets/Scripts/Buttons.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class Buttons : MonoBehaviour
{
    //Suggestive text >:)
    public GameObject Cant;
    public GameObject RedCant;
    public GameObject Cant2;
    public GameObject Cant3;
    public GameObject Cant4;
    public GameObject Cant5;
    public GameObject Cant6;
    public GameObject Cant7;
    //Buttons
    public GameObject levels5;
    public GameObject levels4;
    public GameObject levels3;
    public GameObject levels2;
    //Blockers
    public GameObject Blocker2;
    public GameObject Blocker3;
    public GameObject Blocker4;
    public GameObject Blocker5;
    public int Scream;
    public virtual void Start()
    {
        if (Application.loadedLevelName == "Level")
        {
            if (PlayerPrefs.GetInt("DoneLevel1") == 0)
            {
                this.Blocker2.SetActive(true);
                this.Blocker3.SetActive(true);
                this.Blocker4.SetActive(true);
                this.Blocker5.SetActive(true);
            }
            else
            {
                if (PlayerPrefs.GetInt("DoneLevel2") == 0)
                {
                    this.Blocker3.SetActive(true);
                    this.Blocker4.SetActive(true);
                    this.Blocker5.SetActive(true);
                }
                else
                {
                    if (PlayerPrefs.GetInt("DoneLevel3") == 0)
                    {
                        this.Blocker4.SetActive(true);
                        this.Blocker5.SetActive(true);
                    }
                    else
                    {
                        if (PlayerPrefs.GetInt("DoneLevel4") == 0)
                        {
                            this.Blocker5.SetActive(true);
                        }
                        else
                        {
                            if (PlayerPrefs.GetInt("DoneLevel5") == 0)
                            {
                            }
                        }
                    }
                }
            }
        }
        else
        {
            //nothing shall pass
            if (Application.loadedLevelName == "RLevel") //Reversed Level pack 1
            {
                if (PlayerPrefs.GetInt("DoneRLevel1") == 0)
                {
                    this.Blocker2.SetActive(true);
                    this.Blocker3.SetActive(true);
                    this.Blocker4.SetActive(true);
                    this.Blocker5.SetActive(true);
                }
                else
                {
                    if (PlayerPrefs.GetInt("DoneRLevel2") == 0)
                    {
                        this.Blocker3.SetActive(true);
                        this.Blocker4.SetActive(true);
                        this.Blocker5.SetActive(true);
                    }
                    else
                    {
                        if (PlayerPrefs.GetInt("DoneRLevel3") == 0)
                        {
                            this.Blocker4.SetActive(true);
                            this.Blocker5.SetActive(true);
                        }
                        else
                        {
                            if (PlayerPrefs.GetInt("DoneRLevel4") == 0)
                            {
                                this.Blocker5.SetActive(true);
                            }
                        }
                    }
                }
            }
        }
    }

    public virtual void update()
    {
        Screen.sleepTimeout = SleepTimeout.NeverSleep;
    }

    public virtual void QuitGame()
    {
        Debug.Log("Game is exiting...");
        Application.Quit();
    }

    public virtual void MainMenu()
    {
        Application.LoadLevel("MainMenu");
        Time.timeScale = 1;
    }

    public virtual void Playgame()
    {
        if (PlayerPrefs.HasKey("savedLevel"))
        {
            // there is a saved Level, load that one
            Application.LoadLevel(PlayerPrefs.GetInt("savedLevel"));
        }
        else
        {
            // no saved Level, load the first one
            Application.LoadLevel("Level1");
        }
    }

    public virtual void Tutorial()
    {
        Application.LoadLevel("Tutorial");
    }

    //All the scene selection stuff
    public virtual void levelselectoor()
    {
        Application.LoadLevel("LevelSelect");
    }

    public virtual void levels()
    {
        Application.LoadLevel("Level");
    }

    public virtual void levelsR()
    {
        Application.LoadLevel("RLevel");
    }

    //Level Pack 1
    public virtual void Level1()
    {
        Application.LoadLevel("Level1");
    }

    public virtual void Level2()
    {
        if (PlayerPrefs.GetInt("DoneLevel1") == 1)
        {
            Application.LoadLevel("Level2");
        }
        else
        {
            this.StartCoroutine(this.Warning());
        }
    }

    public virtual void Level3()
    {
        if (PlayerPrefs.GetInt("DoneLevel2") == 1)
        {
            Application.LoadLevel("Level3");
        }
        else
        {
            this.StartCoroutine(this.Warning());
        }
    }

    public virtual void Level4()
    {
        if (PlayerPrefs.GetInt("DoneLevel3") == 1)
        {
            Application.LoadLevel("Level4");
        }
        else
        {
            this.StartCoroutine(this.Warning());
        }
    }

    public virtual void Level5()
    {
        if (PlayerPrefs.GetInt("DoneLevel4") == 1)
        {
            Application.LoadLevel("Level5");
        }
        else
        {
            this.StartCoroutine(this.Warning());
        }
    }

    //Reverse Level pack
    public virtual void RLevel1()
    {
        Application.LoadLevel("RLevel1");
    }

    public virtual void RLevel2()
    {
        if (PlayerPrefs.GetInt("RLevel1") == 1)
        {
            Application.LoadLevel("RLevel2");
        }
        else
        {
            this.StartCoroutine(this.Warning());
        }
    }

    public virtual void RLevel3()
    {
        if (PlayerPrefs.GetInt("RLevel2") == 1)
        {
            Application.LoadLevel("RLevel3");
        }
        else
        {
            this.StartCoroutine(this.Warning());
        }
    }

    public virtual void RLevel4()
    {
        if (PlayerPrefs.GetInt("RLevel3") == 1)
        {
            Application.LoadLevel("RLevel4");
        }
        else
        {
            this.StartCoroutine(this.Warning());
        }
    }

    public virtual void RLevel5()
    {
        if (PlayerPrefs.GetInt("RLevel5") == 1)
        {
            Application.LoadLevel("RLevel5");
        }
        else
        {
            this.StartCoroutine(this.Warning());
        }
    }

    public virtual IEnumerator Warning()
    {
        this.Scream = this.Scream + 1;
        if (this.Scream == 1)
        {
            this.Cant.SetActive(true);
        }
        else
        {
            if (this.Scream == 2)
            {
                this.Cant.SetActive(false);
                this.RedCant.SetActive(true);
            }
            else
            {
                if (this.Scream == 3)
                {
                    this.RedCant.SetActive(false);
                    yield return new WaitForSeconds(0.5f);
                    this.RedCant.SetActive(true);
                }
                else
                {
                    if (this.Scream == 4)
                    {
                        this.Cant2.SetActive(true);
                    }
                    else
                    {
                        if (this.Scream == 5)
                        {
                            this.Cant3.SetActive(true); //ugh
                        }
                        else
                        {
                            if (this.Scream == 6)
                            {
                                this.Cant4.SetActive(true); //making me mad
                            }
                            else
                            {
                                if (this.Scream == 7)
                                {
                                    this.Cant5.SetActive(true); //your asking for it
                                }
                                else
                                {
                                    if (this.Scream == 8)
                                    {
                                        this.Cant5.SetActive(false);
                                        this.Cant4.SetActive(false);
                                        this.Cant3.SetActive(false);
                                        this.Cant2.SetActive(false);
                                        this.RedCant.SetActive(false);
                                        this.Cant6.SetActive(true); //one more time
                                    }
                                    else
                                    {
                                        if (this.Scream == 9)
                                        {
                                            //NUCLEAR
                                            if (PlayerPrefs.GetInt("DoneLevel1") == 0)
                                            {
                                                this.levels5.SetActive(false);
                                                this.levels4.SetActive(false);
                                                this.levels3.SetActive(false);
                                                this.levels2.SetActive(false);
                                            }
                                            else
                                            {
                                                if (PlayerPrefs.GetInt("DoneLevel2") == 0)
                                                {
                                                    this.levels5.SetActive(false);
                                                    this.levels4.SetActive(false);
                                                    this.levels3.SetActive(false);
                                                }
                                                else
                                                {
                                                    if (PlayerPrefs.GetInt("DoneLevel3") == 0)
                                                    {
                                                        this.levels5.SetActive(false);
                                                        this.levels4.SetActive(false);
                                                    }
                                                    else
                                                    {
                                                        if (PlayerPrefs.GetInt("DoneLevel4") == 0)
                                                        {
                                                            this.levels5.SetActive(false);
                                                        }
                                                        else
                                                        {
                                                            if (PlayerPrefs.GetInt("DoneRLevel1") == 0) //Reversed Level Pack 1
                                                            {
                                                                this.levels5.SetActive(false);
                                                                this.levels4.SetActive(false);
                                                                this.levels3.SetActive(false);
                                                                this.levels2.SetActive(false);
                                                            }
                                                            else
                                                            {
                                                                if (PlayerPrefs.GetInt("DoneRLevel2") == 0)
                                                                {
                                                                    this.levels5.SetActive(false);
                                                                    this.levels4.SetActive(false);
                                                                    this.levels3.SetActive(false);
                                                                }
                                                                else
                                                                {
                                                                    if (PlayerPrefs.GetInt("DoneRLevel3") == 0)
                                                                    {
                                                                        this.levels5.SetActive(false);
                                                                        this.levels4.SetActive(false);
                                                                    }
                                                                    else
                                                                    {
                                                                        if (PlayerPrefs.GetInt("DoneRLevel4") == 0)
                                                                        {
                                                                            this.levels5.SetActive(false);
                                                                        }
                                                                    }
                                                                }
                                                            }
                                                        }
                                                    }
                                                }
                                            }
                                            this.Cant6.SetActive(false);
                                            this.Cant7.SetActive(true);
                                            this.Blocker5.SetActive(false);
                                            this.Blocker4.SetActive(false);
                                            this.Blocker3.SetActive(false);
                                            this.Blocker2.SetActive(false);
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }

}